framework/src/outputs/OutputWarehouse.C

```diff
@@ -178,6 +179,15 @@ OutputWarehouse::mooseConsole()
     _console_buffer.clear();
     _console_buffer.str("");
   }
+  else
```

Comment:
There should be an option to disable/enable this feature, in case things are being redirected to a file.